staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go

```diff
@@ -185,48 +185,51 @@ func DefaultVerifyOptions() x509.VerifyOptions {
 
 // CommonNameUserConversion builds user info from a certificate chain using the subject's CommonName
 var CommonNameUserConversion = UserConversionFunc(func(chain []*x509.Certificate) (*authenticator.Response, bool, error) {
```

Comment:
"error", the last value of return, is always set to nil.  Shall we let convertToUserInfo() return some error message and the error is popped up in the call stack?

Reply:
good call. adding.

Reply:
fixed. added error and changed tests accordingly